src/middlewares/authenticationMiddleware.ts

```diff
@@ -0,0 +1,46 @@
+import jwt from 'jsonwebtoken'
+import { Response, Request, NextFunction } from 'express'
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace Express {
+    // eslint-disable-next-line no-shadow
+    interface Request {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      userDetails: any
+    }
+  }
+}
+
+// eslint-disable-next-line import/prefer-default-export
+export const isAuthenticated = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    if (!req.headers.authorization) {
+      return res.status(401).json({ message: 'Please Login' })
+    }
+    const token = req.headers.authorization?.split(' ')[1] || ''
+
+    if (!token) {
+      return res.status(401).json({ message: 'no access token found' })
+    }
+
+    jwt.verify(
+      token,
+      process.env.JWT_SECRET || 'USER-AUTH',
+      (err: unknown, decoded: unknown) => {
+        if (err) {
+          return res.status(401).json({ message: 'Invalid token' })
+        }
+        res.locals.decoded = decoded
+        req.userDetails = decoded
+        next()
+      },
+    )
```

Comment:
```suggestion
   const payload=await decodeToken(token)
   const user=await getUserById(payload.id)
   req.user=user;
   res.locals.decoded
   next()

```
We can use JWT utils instead of hardcording jwt verfication and also  convert this async await for blocking, to make sure that each handler have req.user